tests/macros/test-svginfo.js

```diff
@@ -0,0 +1,496 @@
+                tags: ["SVG", "Element"]
+            }
+        },
+        "view": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'view'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>view</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        }
+    }
+};
+
+
+// Test cases definition
+// ----------------------------------------------------------------------------
+// Each test case is define by:
+// A `title` to make the test understandable by a human behing
+// An `input` which is an Array of parameters that will be passed to the macro
+// An `output` which is the string that the macro should return,
+// A `env` that is overriding the default env variable inside the macro
+//
+// NOTE: we could probably make that more generic by having a single test
+//       runner (see below) and a bunch of JSON files (one per macro) to
+//       describe all the possible inputs and their expected outputs.
+// ----------------------------------------------------------------------------
+
+const TEST_CASE = [{
+    title: 'Test preview mode display (no param, no slug)',
+    input: [],
+    output: '<span style="color:red;">SVG info in preview not available</span>'
+},{
+    title: 'Test unkown SVG element (param: foo)',
+    input: ['foo'],
+    output: '<span style="color:red;">missing</span>'
+},{
+    // Defs allows to test multiple category and a mix of groups and elements for permitted contents
+    title: 'Test explicite slug (param: defs)',
+    input: ['defs'],
+    output: makeExpect(SVG_DATA.elements.defs)
+},{
+    // altGlyphDef allows to test when description is an object rather than a string
+    title: 'Test implicite slug (no param, slug: /en-US/docs/Web/SVG/Element/altGlyphDef)',
+    input: [],
+    output: makeExpect(SVG_DATA.elements.altGlyphDef),
+    env: {
+        slug: URL('en-US', SVG_BASE_SLUG, 'Element', 'altGlyphDef')
+    }
+},{
+    title: 'Test implicite non English slug (no param, slug: /zh-CN/docs/Web/SVG/Element/defs)',
+    input: [],
+    output: makeExpect(SVG_DATA.elements.defs, 'zh-CN'),
+    env: {
+        locale: 'zh-CN',
+        slug: URL('zh-CN', SVG_BASE_SLUG, 'Element', 'defs'),
+    }
+}];
+
+
+// Test runner
+// ----------------------------------------------------------------------------
+
+describeMacro('svginfo', () => {
+    beforeEachMacro((macro) => {
+        // let's make sure we have a clean calls to wiki.getPage
+        macro.ctx.wiki.getPage = sinon.stub();
+
+        Object.keys(MOCK_PAGES['en-US']).forEach((key) => {
+            const { url, data } = MOCK_PAGES['en-US'][key];
+            macro.ctx.wiki.getPage.withArgs(url).returns(data);
+        });
+
+        Object.keys(MOCK_PAGES['zh-CN']).forEach((key) => {
+            const { url, data } = MOCK_PAGES['zh-CN'][key];
+            macro.ctx.wiki.getPage.withArgs(url).returns(data);
+        });
+    });
```

Comment:
This is nicely done, but it's missing the stub for the `wiki.pageExists` call that is made within the `SVGElement` macro (via the call to `template("SVGElement"...` made in one of the paths through the `svginfo` macro). The way the test runs currently, it's making an actual network request via the real `wiki.pageExists`. Something like this worked for me during my testing:
```javascript
    beforeEachMacro((macro) => {
        // let's make sure we have clean calls to wiki.getPage
        macro.ctx.wiki.getPage = sinon.stub();
        macro.ctx.wiki.pageExists = sinon.stub();

        Object.keys(MOCK_PAGES['en-US']).forEach((key) => {
            const { url, data } = MOCK_PAGES['en-US'][key];
            macro.ctx.wiki.getPage.withArgs(url).returns(data);
            macro.ctx.wiki.pageExists.withArgs(url).returns(true);
        });

        Object.keys(MOCK_PAGES['zh-CN']).forEach((key) => {
            const { url, data } = MOCK_PAGES['zh-CN'][key];
            macro.ctx.wiki.getPage.withArgs(url).returns(data);
            macro.ctx.wiki.pageExists.withArgs(url).returns(true);
        });
    });
```

This reminds me that I need to submit a PR that stubs the `request` package within the `macro.ctx` object, so that it throws an error when called. I think that would be the best way to catch all of the attempts to make real network calls, and help us discover what we need to stub.

Reply:
It's been a while since I've worked with the test framework, as I forgot that I had already stubbed the `request` package long ago in https://github.com/mdn/kumascript/pull/240! 🤦‍♂️

So your tests were not making real network calls, sorry! I still think it's best to create and populate the stub for `wiki.pageExists` instead of relying on whatever `wiki.pageExists` returns using the un-populated `request` stub.